### ISSUE TITLE
Add location by QR code

### ISSUE THREAD

**Issue by ohhmm** [RetroShare/RetroShare#238]

**Comment by TheSentry** [comment]
Could you specify what you mean by that exactly? It is not obvious.

**Comment by ohhmm** [comment]
The major idea is tom make contact connection linking easy by showing QR code.
One host generates QR-code and another one reads it.
If ask me about implementation than I'd say QR could contain public key and temporary fast-generated private key for handshake. This would allow to protect from attack possibility to add self to a contact list. After adding by both sides, the temporary key should be removed.

**Comment by TheSentry** [comment]
1) Who reads it? A smartphone? Then how does this get sent to the computer?
2) The QR code would be quite big and hard to scan.
3) How do you transmit the QR code? Do you carry a physical printout with you? If you send it as picture via mail or chat, you can just sent the certificate instead.

Don't get me wrong, I also thought about this. But I haven't heard of a suggestion that would make things easier than certificates.

**Comment by csoler** [comment]
yeah, me too. The QR code option would do nice on smartphones, but RS is a bit too greedy for these devices.

**Comment by kellax-ware** [comment]
The Idea is (IMO) to scan QR code from screen. How it works in WhatsApp-Web/BlackBerryMessenger/Threema. Also you may choose: add QR code as your new location or like new contact. In first case new location has to be approved from desktop.

**Comment by ohhmm** [comment]
1. As I suggest, first certificate generated at a workstation. Then laptop. Then any other computers with camera. But any sequences is good. Also you can connect a friend with QR too as option. Well I could seem creepy to you to show a laptop to other laptop, but believe me, some people think even more creepy is to use flash drive to exchange public certificate for better security then sending it by email. 
2. No. Use ZXing.
3. See first answer.

**Comment by felisucoibi** [comment]
I think this only have sense if there is a retroshare version for phones or tablets (that would be nice to use it as a secure chat and irc with friends), I can't imagine using qr code with a laptop camera...

**Comment by ohhmm** [comment]
Well it makes sense for me not only for phone/tablet, but for laptop too. If functionality would be extremely useful for phones why cut it for laptop? It still optional to use.

**Comment by Osndok** [comment]
I don't understand why the certificate is so large... surely all it would (technically) need for a connection is an initial IP address, port number, and a DHT public key half, no?

**Comment by G10h4ck** [comment]
Short invite is being implemented and will be part of next release 0.6.6

**Comment by G10h4ck** [comment]
The blocking part for 0.6.6 of this issue was implementing short invites. Short invites have been implemented and already merged in master and of course will be part of retroshare 0.6.6, so I remove the `0.6.6 blocker` from this issue.

**Comment by defnax** [comment]
@csoler , qr codes done for retroshare mobile, close this @G10h4ck
